app/src/test/kotlin/StorageUnitTest.kt

```diff
@@ -0,0 +1,59 @@
+package test.kotlin
+
+import android.widget.TextView
+import com.simplemobiletools.camera.R
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import com.simplemobiletools.camera.extensions.config
+import com.simplemobiletools.camera.helpers.DeviceStorageUtil
+import com.simplemobiletools.commons.extensions.isVisible
+import com.simplemobiletools.commons.extensions.value
+import org.junit.* // ktlint-disable no-wildcard-imports
+import org.robolectric.shadows.ShadowStatFs
+
+@RunWith(RobolectricTestRunner::class)
+class StorageUnitTest : KotlinRobolectric() {
+
+    @Test
+    fun testBytesToHuman() {
+        Assert.assertEquals("9.54 Mb", DeviceStorageUtil.bytesToHuman(9999999))
+        Assert.assertEquals("953.67 Mb", DeviceStorageUtil.bytesToHuman(999999999))
+        Assert.assertEquals("9.31 Gb", DeviceStorageUtil.bytesToHuman(9999999999))
+    }
+
+    @Test
+    fun testZeroSpace() {
+
+        // Toggle the space remaining
+        mMainActivity?.config?.spaceRemainingOn = true
+        Assert.assertEquals(true, mMainActivity?.config?.spaceRemainingOn)
+
+        mMainActivity?.setRecordingState(true)
+        Assert.assertEquals(true, mMainActivity?.findViewById<TextView>(R.id.space_remaining)?.isVisible())
+        Assert.assertEquals("0 byte left", mMainActivity?.findViewById<TextView>(R.id.space_remaining)?.value)
+    }
+
+    @Test
+    fun testFreeSpace() {
+        // Toggle the space remaining
+        mMainActivity?.config?.spaceRemainingOn = true
+        Assert.assertEquals(true, mMainActivity?.config?.spaceRemainingOn)
+
+        // Fake Storage Capacity
+        ShadowStatFs.registerStats("/tmp", 100, 1230, 1120)
+
+        mMainActivity?.setRecordingState(true)
+        Assert.assertEquals(true, mMainActivity?.findViewById<TextView>(R.id.space_remaining)?.isVisible())
+        Assert.assertEquals("4.38 Mb left", mMainActivity?.findViewById<TextView>(R.id.space_remaining)?.value)
```

Comment:
This testFreeSpace is failing for me:
Expected: 4.38 Mb left
Actual: 0 byte left

Reply:
I get the same failing test, other than that I looked over and the tests are good. Just fix this failing test and we can merge

Reply:
The tests are working fine on my local machine? Does it tell you something like this: /tmp/robolectric-Method_firstTest6547737288131991820/external-cache ? whats the path of the test?

Reply:
I changed the to use absolute path instead of using /tmp. Try to run it and tell me if it works for you.

Reply:
Works now!